### PR TITLE
ddl: fix drop database bug if the database has partition table 

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
we should drop partition ids if table is partition table.
currently, when drop schema, will drop table by table id no matter whether the table is or not partition table.

### Check List 

Tests 
 - unit test

Related changes

 - Need to cherry-pick to the release branch

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8319)

### CODE REVIEW

**Comment by tiancaiamao** [comment]
Please add test @crazycs520

File: ddl/db_test.go [hunk 1]
```diff
@@ -3607,3 +3607,42 @@ func (s *testDBSuite) TestPartitionAddIndex(c *C) {
 	tk.MustExec("admin check table partition_add_idx")
 	tk.MustExec("drop table partition_add_idx")
 }
+
+func (s *testDBSuite) TestDropSchemaWithPartitionTable(c *C) {
```
Comment (winkyao) [inline ddl/db_test.go]: add more tests to ensure the data of the partitioned table is dropped.

File: ddl/schema.go [hunk 2]
```diff
@@ -128,7 +128,11 @@ func onDropSchema(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 func getIDs(tables []*model.TableInfo) []int64 {
 	ids := make([]int64, 0, len(tables))
 	for _, t := range tables {
-		ids = append(ids, t.ID)
+		if t.GetPartitionInfo() == nil {
```
Comment (tiancaiamao) [inline ddl/schema.go]: The partitioned table itself should included. @crazycs520 @ciscoxll 

```
ids = append(ids, t.ID)
if t.GetPartitionInfo() != nil {
    ids = append(ids, getPartitionIDs(t)...)
}
```

Comment (crazycs520) [reply ddl/schema.go]: done.

**Comment by tiancaiamao** [comment]
LGTM @winkyao

**Pull request merged**